### PR TITLE
reduce `GatherSlice` host memory footprint

### PR DESCRIPTION
**Pull request by psychocoderHPC**
free mpi gatherv memory after usage

related to #1279

**Tests:**

- [x] 3D LWFA with sliding window (checked png images)
- 2D LWFA with sliding window 
  - [x] checked png images
  - [x] check memory usage with `top -d 0.5`

### CODE REVIEW

File: src/picongpu/include/plugins/output/GatherSlice.hpp [hunk 1]
```diff
@@ -188,6 +188,7 @@ struct GatherSlice
                 insertData(dstBox, srcBox, head->node.offset, head->node.maxSize);
             }
 
+            __deleteArray(fullData);
```
Comment (erikzenker) [inline src/picongpu/include/plugins/output/GatherSlice.hpp]: Why using __deleteArray when the rest of the code uses delete[] ?

Comment (psychocoderHPC) [inline src/picongpu/include/plugins/output/GatherSlice.hpp]: `__deleteArray()` is a PMacc macro which sets the pointer after it is deleted to `NULL`. This macro is newer than tthis class, therefore it is currently not used in the destructor or at other parts of this class.

**Comment by erikzenker** [comment]
fullData does not need to be a class member if it is just used in the [operator()](https://github.com/psychocoderHPC/picongpu/blob/topic-reduceMemoryFootprintInGatherSlice/src/picongpu/include/plugins/output/GatherSlice.hpp#L116) methods. Or do you want to reuse it somewhere else, then you should ne delete it at the end. Furthermore, why not use the RAII principle of C++ and declare fullData as std::vector and leave the cleaup to the destructor ...just IHO as a c++ programmer.

File: src/picongpu/include/plugins/output/GatherSlice.hpp (hunk 1 above)
Comment (erikzenker) [inline src/picongpu/include/plugins/output/GatherSlice.hpp]: I there some policy about replacing delete[] by __deleteArray when a file is touched ?

Comment (ax3l) [inline src/picongpu/include/plugins/output/GatherSlice.hpp]: not yet, but if you want you can create a PR.
also, the macros need a little update, see #532

**Pull request merged**